metadata_backend/api/middlewares.py

```diff
@@ -121,3 +129,52 @@ def _json_problem(exception: web.HTTPError, url: URL, _type: str = "about:blank"
         escape_forward_slashes=False,
     )
     return body
+
+
+async def create_session_with_token(req: web.Request) -> aiohttp_session.Session:
+    """Create an authenticated session for the duration of a single request by using an access token.
+
+    :param req: request object containing database connection and request headers
+    :returns: session cookie
+    """
+    session = await aiohttp_session.get_session(req)
+
+    # Get bearer token from Authorization header
+    header = req.headers.get("Authorization", "")
+    header_parts: List[str] = header.split(" ")
+    if len(header_parts) == 2:
+        if header_parts[0] != "Bearer":
+            return session  # current empty session
+    else:
+        return session  # current empty session
+
+    # Get path to userinfo from OIDC config
+    aai = ServiceHandler(base_url=URL(aai_config["oidc_url"].rstrip("/")))
+    aai.service_name = "aai_for_single_session_token_auth"
+    oidc_config = await aai._request(method="GET", path="/.well-known/openid-configuration")
+    if not oidc_config or "userinfo_endpoint" not in oidc_config:
+        return session  # current empty session
+
+    # Validate token by sending it to AAI userinfo and getting back user profile
+    headers = {"Authorization": f"Bearer {header_parts[1]}"}
+    # aai = ServiceHandler(base_url=URL(oidc_config["userinfo_endpoint"]), http_client_headers=headers)
+    aai.base_url = oidc_config["userinfo_endpoint"]
+    aai.http_client_headers = headers
+    userinfo = await aai._request(method="GET")
+    if not userinfo:
+        return session  # current empty session
+
+    # Parse user profile data from userinfo endpoint response, these steps can raise 401
+    single_session: AccessHandler = AccessHandler(aai_config)
+    user_data: Dict[str, Union[List[Dict[str, str]], str]] = await single_session._create_user_data(userinfo)
```

Comment:
Should we have a type alias for this, as it seems to repeat? 
e.g. `UserData = Dict[str, Union[List[Dict[str, str]], str]]`